lib/galaxy/tools/wrappers.py

```diff
@@ -79,6 +79,8 @@ def _get_cast_value(self, other):
         # For backward compatibility, allow `$wrapper != ""` for optional non-text param
         if self.input.optional and self.value is None:
             if isinstance(other, string_types):
+                if other == '':
+                    return ''
```

Comment:
This change doesn't seem necessary for the mentioned optional `select` use case, or am I missing something?

Reply:
I think this is more correct, otherwise $wrapper == "" is not going to work ()

Reply:
That also works because `str(self)` returns `""` when `self.value` is None, due to `to_param_dict_string()` implementation.